internal/libyaml/scanner.go

```diff
@@ -599,19 +593,20 @@ func (parser *Parser) readLine(s []byte) []byte {
 }
 
 // Scan gets the next token.
-func (parser *Parser) Scan(token *Token) bool {
+func (parser *Parser) Scan(token *Token) error {
 	// Erase the token object.
 	*token = Token{} // [Go] Is this necessary?
 
 	// No tokens after STREAM-END or error.
-	if parser.stream_end_produced || parser.ErrorType != NO_ERROR {
-		return true
+	if parser.stream_end_produced || parser.errored {
+		return io.EOF
 	}
 
 	// Ensure that the tokens queue contains enough tokens.
 	if !parser.token_available {
-		if !parser.fetchMoreTokens() {
-			return false
+		if err := parser.fetchMoreTokens(); err != nil {
+			parser.errored = true
```

Comment:
Is there a need for this `errored` variable?

If we do, I would prefer an `hadError`, `errorOccured`, or variants